ToolChain/Assembler/Parser/Parser.py

```diff
@@ -517,6 +429,68 @@ def _getInstructionCodeAndFormUsingPossibleCodes(self,
 
         return instructionCode, instructionForm
 
+    def _parseIndicatorData(self, line, buildInstruction):
+
+        '''
+        Helper method to segregate the evaluateAndExtractInfoFromLine method's operations.
+        This is called if the first argument of a line isn't an instruction. Cases could be:
+            -Global memory reference label
+            -Local memory reference label
+            -Comment
+            -.dataAlpha, .dataNumeric, or .dataMemref
+
+        :param line:
+        :param buildInstruction:
+        :return:
+        '''
+
+        sline = line.split()
+
+        if EXPORTED_REFERENCE_INDICATOR in buildInstruction.operationMnemonic:
+            # Global memory reference which can be used by other files. This is a special case
+            buildInstruction.instructionCode = sline[0].upper() + " "
+            buildInstruction.instructionCode += sline[1].replace(MEMORY_REFERENCE_INDICATORS, "").upper()
+
+            return
+
+        if DATA_NUMERIC_INDICATOR in buildInstruction.operationMnemonic:
+            # Treat this as a numeric field
+            # The numeric value will be treated as a source immediate value in the instruction object
+            numericValue = self.translateTextImmediateToImmediate(sline[1])
+            buildInstruction.sourceImmediate = numericValue
+            buildInstruction.instructionLength = 4
+            buildInstruction.instructionCode = sline[0].upper()
+
+            return
+
+        if DATA_ALPHA_INDICATOR in buildInstruction.operationMnemonic:
+            # Treat this as an alphabetic field
+            # The alpha value will be put in the source immediate value as well
+            # We need to rebuild the alpha into a byte string to be read by the VM
+            # We want to preserve spaces in the original string, so we limit the split to the first argument
+
+            textline = line.split(maxsplit=1)
+            sline = line.split()
+            alpha = b"" + textline[1].encode("utf-8") + b" "
```

Comment:
I don't think the extra splitting of the line is worth this simplification. While I think this is nice, it doesn't really add much benefit. 

If you want to simplify this, using the original fragment:
```python3
alpha = b""	
alpha += [part.encode("utf-8") + b" " for part in line[1:]]	
alpha = alpha[:-1] + b"\x00"  # Remove trailing space	 Strings are null terminated	
buildInstruction.sourceImmediate = alpha	
buildInstruction.instructionLength = len(alpha)	
buildInstruction.instructionCode = line[0].upper()
```

But I'm weary even of my own suggestion as needless wizardry. There are plenty of zip and joins that can accomplish this as well.